MapboxNavigation/SpeechSynthesizing.swift

```diff
@@ -0,0 +1,111 @@
+    func prepareIncomingSpokenInstructions(_ instructions: [SpokenInstruction])
+    
+    /// A request to vocalize the instruction
+    /// - parameter instruction: an instruction to be vocalized
+    /// - parameter legProgress: current leg progress, corresponding to the instruction
+    ///
+    /// This method is not guaranteed to be synchronous or asynchronous. When vocalizing is finished, `voiceController(_ voiceController: SpeechSynthesizing, didSpeak instruction: SpokenInstruction, with error: SpeechError?)` should be called.
+    func speak(_ instruction: SpokenInstruction, during legProgress: RouteLegProgress)
+    
+    /// Tells synthesizer to stop current vocalization in a graceful manner
+    func stopSpeaking()
+    /// Tells synthesizer to stop current vocalization immediately
+    func interruptSpeaking()
+}
+
+/**
+The `SpeechSynthesizingDelegate` protocol defines methods that allow an object to respond to significant events related to spoken instructions.
+ */
+public protocol SpeechSynthesizingDelegate: class, UnimplementedLogging {
+    /**
+     Called when the speech synthesizer encountered an error during processing, but may still be able to speak the instuction.
+     - parameter speechSynthesizer: The voice controller that experienced the failure.
+     - parameter error: An error explaining the failure and its cause.
+     */
+    func speechSynthesizer(_ speechSynthesizer: SpeechSynthesizing, encounteredError error: SpeechError)
+    
+    /**
+    Called when the speech synthesizer finished pronouncing the instruction or encountered an error, which it cannot recover from.
+    - parameter speechSynthesizer: The speech synthesizer performing the action.
+    - parameter instruction: The spoken instruction pronounced or attempted to pronounce
+    - parameter error: An error explaining the failure and its cause if any
+    */
+    func speechSynthesizer(_ speechSynthesizer: SpeechSynthesizing, didSpeak instruction: SpokenInstruction, with error: SpeechError?)
+    
+    /**
+     Called when one spoken instruction interrupts another instruction currently being spoken.
+     
+     - parameter speechSynthesizer: The speech synthesizer that experienced the interruption.
+     - parameter interruptedInstruction: The spoken instruction currently in progress that has been interrupted.
+     - parameter interruptingInstruction: The spoken instruction that is interrupting the current instruction.
+     */
+    func speechSynthesizer(_ speechSynthesizer: SpeechSynthesizing, didInterrupt interruptedInstruction: SpokenInstruction, with interruptingInstruction: SpokenInstruction)
+    
+    /**
+     Called when a spoken is about to speak. Useful if it is necessary to give a custom instruction instead. Noting, changing the `distanceAlongStep` property on `SpokenInstruction` will have no impact on when the instruction will be said.
```

Comment:
Maybe better to add `spoken instruction`? Not a native English speaker myself though :)
```suggestion
     Called when a spoken instruction is about to be spoken. Useful if it is necessary to give a custom instruction instead. Note that changing the `distanceAlongStep` property on `SpokenInstruction` will have no impact on when the instruction will be said.
```